src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java

```diff
@@ -100,14 +114,15 @@ void start()
         // The run loop for the manager thread
         Runnable runnable = new WrappedRunnable()
         {
-            public void runMayThrow() throws Exception
+            public void runMayThrow()
             {
                 while (!shutdown)
                 {
                     try
                     {
                         assert availableSegment == null;
-                        logger.trace("No segments in reserve; creating a fresh one");
+                        if (logger.isTraceEnabled())
```

Comment:
nit: Is there an expensive calculation to protect here?

Reply:
As far as I know it is a good practice to put always logger.trace in such an if statement. 
TRACE-level logging tends to to be massively verbose, and can take up a significant proportion of CPU time, even if it's ultimately not actually logged to the file. 
http://logging.apache.org/log4j/1.2/manual.html#performance
Some people do It also for drug and info, I see it for trace in the cassandra code